src/_pytest/monkeypatch.py

```diff
@@ -147,19 +147,19 @@ def test_partial(monkeypatch):
         finally:
             m.undo()
 
-    @overload
-    def setattr(
+    @overload  # noqa: A003
```

Comment:
Ditto here about the decorator... not sure why it complains about decorator usage.

Reply:
resolving as per my comment below your review. Please reopen if you disagree